pkg/runtime/runtime.go

```diff
@@ -695,6 +690,110 @@ func (a *DaprRuntime) beginPubSub(subscribeCtx context.Context, name string, ps
+			if !allowed {
+				log.Warnf("Actor Pubsub: subscription to topic %s on pubsub %s is not allowed", actorConfigSubscriptions.Topic, name)
+				continue
+			}
+			log.Debugf("Actor Pubsub: subscribing to topic=%s on pubsub=%s", actorConfigSubscriptions.Topic, name)
+
+			if subscribeError := ps.Subscribe(subscribeCtx, pubsub.SubscribeRequest{
+				Topic:    actorConfigSubscriptions.Topic,
+				Metadata: make(map[string]string),
+			}, func(ctx context.Context, msg *pubsub.NewMessage) error {
+				// Match Config with message
+				for _, actorPubSubSubscription := range a.appConfig.Pubsub {
+					// Match Topic with config Topic
+					if actorPubSubSubscription.Topic == msg.Topic && actorPubSubSubscription.PubSubName == name {
+						var cloudEvent map[string]interface{}
+						err := json.Unmarshal(msg.Data, &cloudEvent)
+						if err != nil {
+							log.Errorf("Actor Pubsub: error deserializing cloud event in pubsub %s and topic %s: %s", name, msg.Topic, err)
+							if configured, dlqErr := a.sendToDeadLetterIfConfigured(name, msg); configured && dlqErr == nil {
+								// dlq has been configured and message is successfully sent to dlq.
+								diag.DefaultComponentMonitoring.PubsubIngressEvent(ctx, pubsubName, strings.ToLower(string(pubsub.Drop)), msg.Topic, 0)
+								return nil
+							}
+							diag.DefaultComponentMonitoring.PubsubIngressEvent(ctx, pubsubName, strings.ToLower(string(pubsub.Retry)), msg.Topic, 0)
+							return err
+						}
+
+						if pubsub.HasExpired(cloudEvent) {
+							log.Debugf("Actor Pubsub: dropping expired pub/sub event %v as of %v", cloudEvent[pubsub.IDField], cloudEvent[pubsub.ExpirationField])
+							diag.DefaultComponentMonitoring.PubsubIngressEvent(ctx, pubsubName, strings.ToLower(string(pubsub.Drop)), msg.Topic, 0)
+							a.sendToDeadLetterIfConfigured(name, msg)
+							return nil
+						}
+						// Actor calling if ActorType is correct
+						if cloudEvent[pubsub.ActorTypeField] == "" || cloudEvent[pubsub.ActorTypeField] == actorPubSubSubscription.ActorType {
+							err := a.callActorPubsub(ctx, cloudEvent, actorPubSubSubscription)
+							if err != nil {
+								return err
+							}
+							return nil
+						}
+					}
+				}
+				return nil
+			}); subscribeError != nil {
+				log.Error("Actor Pubsub: Failed to subscribe to topic %s", actorConfigSubscriptions.Topic)
+				return subscribeError
+			}
+		}
+	}
+	return nil
+}
+
+// actorCallPubsub calls the actor that was declared in the cloudevent or in the app.Config
+func (a *DaprRuntime) callActorPubsub(ctx context.Context, cloudEvent map[string]interface{}, actorPubSubSubscription config.PubSubConfig) error {
+	sendData, _ := json.Marshal(cloudEvent[pubsub.DataField])
+	if cloudEvent[pubsub.ActorIdField] == "" {
+		// Actor id is not in the cloudevent
+		if actorPubSubSubscription.ActorIDDataAttribute == "" {
+			log.Debug("Actor Pubsub: ActorID and ActorIDDataAttribute not declared in cloud event id - %s", cloudEvent[pubsub.IDField])
```

Comment:
Log Warning here. Users probably want to know the event was ignored.

Should these go to the DLQ instead?